src/lib.rs

```diff
@@ -31,31 +56,42 @@ fn default_result<T>() -> Result<T, Error> {
 ///
 /// ```
 /// let saddr = "127.0.0.1:443".parse().unwrap();
-/// let mtu = mtu::interface_mtu(&saddr).unwrap();
-/// println!("MTU towards {saddr:?} is {mtu}");
+/// let mtu = mtu::interface_mtu((None, saddr)).unwrap();
+/// println!("MTU for {saddr:?} is {mtu}");
 /// ```
 ///
 /// # Errors
 ///
 /// This function returns an error if the local interface MTU cannot be determined.
-pub fn interface_mtu(remote: &SocketAddr) -> Result<usize, Error> {
+pub fn interface_mtu<A>(addrs: A) -> Result<usize, Error>
+where
+    SocketAddrs: From<A>,
+    A: std::marker::Copy + std::fmt::Debug,
+{
     #[cfg(any(target_os = "macos", target_os = "linux", target_os = "windows"))]
     #[allow(unused_assignments)] // Yes, res is reassigned in the platform-specific code.
     let mut res = default_result();
 
     #[cfg(any(target_os = "macos", target_os = "linux", target_os = "windows"))]
     {
-        // Make a new socket that is connected to the remote address. We use this to learn which
-        // local address is chosen by routing.
-        let socket = UdpSocket::bind((
-            if remote.is_ipv4() {
-                IpAddr::V4(Ipv4Addr::UNSPECIFIED)
-            } else {
-                IpAddr::V6(Ipv6Addr::UNSPECIFIED)
-            },
-            0,
-        ))?;
-        socket.connect(remote)?;
+        let local = match addrs.into() {
+            SocketAddrs::Local(local) | SocketAddrs::Both((local, _)) => local,
+            SocketAddrs::Remote(remote) => SocketAddr::new(
+                if remote.is_ipv4() {
+                    IpAddr::V4(Ipv4Addr::UNSPECIFIED)
+                } else {
+                    IpAddr::V6(Ipv6Addr::UNSPECIFIED)
+                },
+                0,
+            ),
+        };
+        let socket = UdpSocket::bind(local)?;
+        match addrs.into() {
```

Comment:
I'm surprised that you weren't told that this was uncool.  I guess that it is that `Copy` implementation.

```suggestion
        match addrs {
```